internal/configs/config.go

```diff
@@ -955,8 +985,193 @@ func (c *Config) TransformForTest(run *TestRun, file *TestFile) (func(), hcl.Dia
+			}
+			// It could be a warning, but for the sake of consistent UX let's make it an error
+			diags = append(diags, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  fmt.Sprintf("Unsupported `%v` target in `%v` block", targetMode, blockName),
+				Detail: fmt.Sprintf("Target `%v` is `%v` block itself and cannot be overriden with `%v`.",
+					overrideRes.TargetParsed, targetMode, blockName),
+				Subject: overrideRes.Target.SourceRange().Ptr(),
+			})
+		}
+
+		res.IsOverridden = true
+		res.OverrideValues = overrideRes.Values
+	}
+
+	return func() {
+		// Reset all the overriden resources.
+		for _, o := range run.OverrideResources {
+			m := c.Root.Descendent(o.TargetParsed.Module)
+			if m == nil {
+				continue
+			}
+
+			res := m.Module.ResourceByAddr(o.TargetParsed.Resource)
+			if res == nil {
+				continue
+			}
+
+			res.IsOverridden = false
+			res.OverrideValues = nil
+		}
+	}, diags
+}
+
+func (c *Config) transformOverridenModulesForTest(run *TestRun, file *TestFile) (func(), hcl.Diagnostics) {
+	modules, diags := mergeOverridenModules(run.OverrideModules, file.OverrideModules)
+
+	for _, overrideMod := range modules {
+		targetConfig := c.Root.Descendent(overrideMod.TargetParsed)
+		if targetConfig == nil {
+			diags = append(diags, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  fmt.Sprintf("Module not found: %v", overrideMod.TargetParsed),
+				Detail:   "Target points to an undefined module. Please, ensure module exists.",
+				Subject:  overrideMod.Target.SourceRange().Ptr(),
+			})
+			continue
+		}
+
+		for overrideKey := range overrideMod.Outputs {
+			if _, ok := targetConfig.Module.Outputs[overrideKey]; !ok {
+				diags = append(diags, &hcl.Diagnostic{
+					Severity: hcl.DiagWarning,
+					Summary:  fmt.Sprintf("Output not found: %v", overrideKey),
+					Detail:   "Specified output to override is not present in the module and will be ignored.",
+					Subject:  overrideMod.Target.SourceRange().Ptr(),
+				})
+			}
+		}
+
+		targetConfig.Module.IsOverridden = true
+
+		for key, output := range targetConfig.Module.Outputs {
+			output.IsOverridden = true
+
+			// Override outputs are optional so it's okay to set IsOverridden with no OverrideValue.
+			if v, ok := overrideMod.Outputs[key]; ok {
+				output.OverrideValue = &v
+			}
+		}
+	}
+
+	return func() {
+		for _, overrideMod := range run.OverrideModules {
+			targetConfig := c.Root.Descendent(overrideMod.TargetParsed)
+			if targetConfig == nil {
+				continue
+			}
+
+			targetConfig.Module.IsOverridden = false
+
+			for _, output := range targetConfig.Module.Outputs {
+				output.IsOverridden = false
+				output.OverrideValue = nil
+			}
+		}
+	}, diags
+}
+
+func mergeOverridenResources(runResources, fileResources []*OverrideResource) ([]*OverrideResource, hcl.Diagnostics) {
+	// runAddrs is a unique set of resource addresses in run block.
+	// It's already validated for duplicates previously.
+	runAddrs := make(map[string]struct{})
```

Comment:
The same as in `mergeOverridenModules`: It stands for resource addresses in run block. It could be `runResourceAddrs` or `addrsInRun` if just `runAddrs` is not clear

Reply:
7e162882a7c5616aa71cd116cc67197bd5d997d7